embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/FailedApiCallsPerEndpoint.kt

```diff
@@ -0,0 +1,80 @@
+package io.embrace.android.embracesdk.comms.delivery
+
+import io.embrace.android.embracesdk.comms.api.EmbraceApiService.Companion.Endpoint
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * A map containing a list of failed API calls per endpoint.
+ */
+internal class FailedApiCallsPerEndpoint {
+
+    private val failedApiCallsMap = ConcurrentHashMap<Endpoint, DeliveryFailedApiCalls>()
+
+    /**
+     * Adds a failed API call in the corresponding endpoint's list.
+     */
+    fun add(endpoint: Endpoint, failedApiCall: DeliveryFailedApiCall) {
+        val failedApiCallsForEndpoint = failedApiCallsMap.getOrPut(endpoint) { DeliveryFailedApiCalls() }
+        failedApiCallsForEndpoint.add(failedApiCall)
+    }
+
+    /**
+     * Returns the list of failed API calls for the corresponding endpoint.
+     */
+    fun get(endpoint: Endpoint): DeliveryFailedApiCalls? {
```

Comment:
Do we need to expose this? Is it possible for the interface of the object to just be `add()` `getNext()`, `clear()`, and `hasAnyFailedApiCalls()`, leaving the limits, counts, etc. be internal to the object?

Basically, the retry manager should ask: "do you have a request for me to retry?" and this will just return if it has one. Not sure if it needs to expose more than that right now before we add the 429 stuff...

Reply:
Done. I had to update a bit the code in the retryManager to avoid using the count. I leaved just the methods you are mentioning.